documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -267,49 +280,73 @@ If a consumer group or standalone consumer is inactive and commits no offsets du
 
 If the amount of data returned in a single fetch request is large,
 a timeout might occur before the consumer has processed it.
 In this case, you can lower `max.partition.fetch.bytes` or increase `session.timeout.ms`.
 
-== Minimizing the impact of rebalances
+== Minimizing the impact of rebalances 
 
+Rebalances in Kafka consumer groups can introduce latency and reduce throughput, impacting overall service performance. 
 The rebalancing of a partition between active consumers in a group is the time it takes for the following to take place:
 
 * Consumers to commit their offsets
 * The new consumer group to be formed
 * The group leader to assign partitions to group members
 * The consumers in the group to receive their assignments and start fetching
 
-The rebalancing process can increase the downtime of a service, particularly if it happens repeatedly during a rolling restart of a consumer group cluster.
+Rebalances are triggered by changes in consumer health, network issues, configuration updates, and scaling events.
+This process can increase service downtime, especially if it occurs frequently, such as during rolling restarts of a consumer group cluster. 
+To minimize the impact of rebalances, consider the following strategies and configurations:
+
+Static membership:: Assign a unique identifier (`group.instance.id`) to each consumer instance. 
+Static membership introduces persistence so static consumers retain partition assignments across restarts, reducing unnecessary rebalances. 
 
-In this situation, you can introduce _static membership_ by assigning a unique identifier (`group.instance.id`) to each consumer instance within the group.
-Static membership uses persistence so that a consumer instance is recognized during a restart after a session timeout.
-Consequently, the consumer maintains its assignment of topic partitions, reducing unnecessary rebalancing when it rejoins the group after a failure or restart.
- 
-Additionally, adjusting the `max.poll.interval.ms` configuration can prevent rebalances caused by prolonged processing tasks, allowing you to specify the maximum interval between polls for new messages.
-Use the `max.poll.records` property to cap the number of records returned from the consumer buffer during each poll. 
-Reducing the number of records allows the consumer to process fewer messages more efficiently.  
-In cases where lengthy message processing is unavoidable, consider offloading such tasks to a pool of worker threads. 
-This parallel processing approach prevents delays and potential rebalances caused by overwhelming the consumer with a large volume of records.
+Poll intervals and record limits:: 
+* Use the `max.poll.interval.ms` property to prevent rebalances caused by prolonged processing tasks by setting the maximum interval between polls.
+* Use the `max.poll.records` property to limit the number of records returned during each poll. 
```

Comment:
I'd recommend tuning this config first. In theory you just need to adjust this config. 
Another option is to pause some partitions to retrieve less records.

In practice I favor the 2 options above as when you start adjusting the various timeouts and intervals you start changing the behavior in case of failures.